src/commands.ts

```diff
@@ -408,32 +410,34 @@ export function viewDocumentation() {
     });
 }
 
-// rename to getJSFromTS
-// export async function uploadJSFromTS(sdsConnection: SDSConnection, textDocument: vscode.TextDocument): Promise<void> {
-//     return new Promise<void>((resolve, reject) => {
-
-//         if(!textDocument || '.ts' !== path.extname(textDocument.fileName)) {
-//             reject('No active ts script');
-
-//         } else {
-//             let shortName = '';
-//             let scriptSource = '';
-
-//             shortName = path.basename(textDocument.fileName, '.ts');
-//             let tsname:string = textDocument.fileName;
-//             let jsname:string = tsname.substr(0, tsname.length - 3) + ".js";
-//             //let tscargs = ['--module', 'commonjs', '-t', 'ES6'];
-//             let tscargs = ['-t', 'ES5', '--out', jsname];
-//             let retval = tsc.compile([textDocument.fileName], tscargs, null, function(e) { console.log(e); });
-//             scriptSource = retval.sources[jsname];
-//             console.log("scriptSource: " + scriptSource);
-
-//             sdsAccess.uploadScript(sdsConnection, shortName, scriptSource).then((value) => {
-//                 vscode.window.setStatusBarMessage('uploaded: ' + shortName);
-//                 resolve();
-//             }).catch((reason) => {
-//                 reject(reason);
-//             });
-//         }
-//     });
-// }
+
+
+
+
+export async function uploadJSFromTS(loginData: nodeDoc.LoginData, textDocument: vscode.TextDocument): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+
+        if (!textDocument || '.ts' !== path.extname(textDocument.fileName)) {
+            reject('No active ts script');
```

Comment:
Consider reject with an Error instance and immediately leave the function with an early return here. E.g.

```ts
return reject(new Error('No active ts script'));
```
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject and https://stackoverflow.com/a/32536083/104274.